tests/common/utils/test_prompt_custom_dir.py

```diff
@@ -0,0 +1,43 @@
+import os
+import shutil
+from unittest.mock import MagicMock
+import click
+from _pytest.capture import CaptureFixture
+
+
+def test_generated_rules_custom_directories(capsys: CaptureFixture[str]):
```

Comment:
I separated this into a different unit test file to avoid messing with a bunch of chaos related to monkeypatching, pytest fixtures, and import orders.